server/liveblog/syndication/utils.py

```diff
@@ -0,0 +1,281 @@
+            },
+            'caption': meta.get('caption', ''),
+            'credit': meta.get('credit', '')
+        },
+        'text': text
+    }
+
+
+def get_producer_post_id(in_syndication, post_id):
+    """Helps to denormalize syndication producer blog post data and provide unique value for producer_post_id field."""
+    return '{}:{}:{}:{}'.format(
+        in_syndication['blog_id'],
+        in_syndication['producer_id'],
+        in_syndication['producer_blog_id'],
+        post_id
+    )
+
+
+def extract_producer_post_data(post, fields=('_id', '_updated', 'highlight', 'sticky', 'post_status')):
+    """Extract only useful data from original producer blog post."""
+    return {key: post[key] for key in fields}
+
+
+def get_post_creator(post):
+    """Get publisher/author from consumer post."""
+    ref = None
+    for group in post['groups']:
+        if group['id'] == 'main':
+            ref = group['refs'][0]
+    if ref:
+        items_service = get_resource_service('blog_items')
+        item = items_service.find_one(req=None, guid=ref['residRef'])
+
+        if item:
+            return item.get('original_creator')
+
+
+def create_syndicated_blog_post(producer_post, items, in_syndication):
+    """Create syndicted blog post data using producer post, fetched items and incoming syndication."""
+    post_items = []
+    for item in items:
+        meta = item.pop('meta')
+        if item['item_type'] == 'image':
+            item = _fetch_and_create_image_item(
+                renditions=meta['media']['renditions'],
+                caption=meta['caption'],
+                credit=meta['credit']
+            )
+        item['blog'] = in_syndication['blog_id']
+        post_items.append(item)
+
+    items_service = get_resource_service('blog_items')
+    item_ids = items_service.post(post_items)
+    item_refs = []
+    for i, item_id in enumerate(item_ids):
+        item_refs.append({
+            'residRef': str(item_id)
+        })
+
+    auto_publish = in_syndication.get('auto_publish', False)
+    if auto_publish:
+        post_status = 'open'
+    else:
+        post_status = 'submitted'
+
+    producer_post_id = get_producer_post_id(in_syndication, producer_post['_id'])
+    new_post = {
+        'blog': in_syndication['blog_id'],
+        'groups': [
+            {
+                'id': 'root',
+                'role': 'grpRole:NEP',
+                'refs': [
+                    {'idRef': 'main'}
+                ]
+            },
+            {
+                'id': 'main',
+                'role': 'grpRole:Main',
+                'refs': item_refs
+            }
+        ],
+        'highlight': False,
+        'sticky': False,
+        'syndication_in': in_syndication['_id'],
+        'particular_type': 'post',
+        'post_status': post_status,
+        'producer_post_id': producer_post_id,
+        # TODO: python-eve doesn't allow to replace _updated value, as post/put/patch method force it to now()
```

Comment:
ok, but we should keep that in mind and open an issue on python-eve if necessary